Source/WebCore/page/Quirks.cpp

```diff
@@ -2254,6 +1805,1010 @@ URL Quirks::topDocumentURL() const
+{
+    if (quirksDomainString != "youtube.com"_s)
+        return;
+
+    UNUSED_PARAM(quirksURL);
+    UNUSED_PARAM(documentURL);
+    quirksData.isYouTube = true;
+    // youtube.com https://bugs.webkit.org/show_bug.cgi?id=195598
+    quirksData.hasBrokenEncryptedMediaAPISupportQuirk = true;
+    // youtube.com rdar://135886305
+    quirksData.needsScrollbarWidthThinDisabledQuirk = true;
+    // youtube.com rdar://66242343
+    quirksData.needsVP9FullRangeFlagQuirk = true;
+#if PLATFORM(IOS_FAMILY)
+    // YouTube.com does not provide AirPlay controls in fullscreen
+    // (Ref: rdar://121471373)
+    quirksData.shouldDisableElementFullscreen = PAL::currentUserInterfaceIdiomIsSmallScreen();
+    auto topDocumentHost = quirksURL.host();
+    if (topDocumentHost == "www.youtube.com"_s) {
+        // www.youtube.com rdar://52361019
+        quirksData.needsYouTubeMouseOutQuirk = true;
+        // youtube.com rdar://49582231
+        quirksData.needsYouTubeOverflowScrollQuirk = true;
+    }
+#endif
+#if PLATFORM(IOS) || PLATFORM(VISION)
+    // youtube.com: rdar://110097836
+    quirksData.shouldSilenceResizeObservers = true;
+#endif
+}
+
+static void handleZillowQuirks(QuirksData& quirksData, const URL& quirksURL, const String& quirksDomainString, const URL& documentURL)
+{
+    if (quirksDomainString != "zillow.com"_s)
+        return;
+
+    UNUSED_PARAM(documentURL);
+    // zillow.com rdar://53103732
+    auto topDocumentHost = quirksURL.host();
+    quirksData.shouldAvoidScrollingWhenFocusedContentIsVisibleQuirk = topDocumentHost == "www.zillow.com"_s;
+#if PLATFORM(IOS) || PLATFORM(VISION)
+    // rdar://110097836
+    quirksData.shouldSilenceResizeObservers = true;
+#endif
+}
+
+static void handleZoomQuirks(QuirksData& quirksData, const URL& quirksURL, const String& quirksDomainString, const URL& documentURL)
+{
+    if (quirksDomainString != "zoom.us"_s)
+        return;
+
+    UNUSED_PARAM(quirksURL);
+    UNUSED_PARAM(documentURL);
+    quirksData.isZoom = true;
+    // zoom.com https://bugs.webkit.org/show_bug.cgi?id=223180
+    quirksData.shouldAutoplayWebAudioForArbitraryUserGestureQuirk = true;
+#if ENABLE(MEDIA_STREAM)
+    // zoom.us rdar://118185086
+    quirksData.shouldDisableImageCaptureQuirk = true;
+#endif
+}
+
+void Quirks::determineRelevantQuirks()
+{
+    RELEASE_ASSERT(m_document);
+    m_quirksData = { };
+
+#if PLATFORM(IOS_FAMILY)
+    m_quirksData.shouldDisableLazyIframeLoadingQuirk = !linkedOnOrAfterSDKWithBehavior(SDKAlignedBehavior::NoUNIQLOLazyIframeLoadingQuirk) && WTF::IOSApplication::isUNIQLOApp();
+    // DOFUS Touch app (rdar://112679186)
+    m_quirksData.needsResettingTransitionCancelsRunningTransitionQuirk = !linkedOnOrAfterSDKWithBehavior(SDKAlignedBehavior::ResettingTransitionCancelsRunningTransitionQuirk) && WTF::IOSApplication::isDOFUSTouch();
+    m_quirksData.shouldEnableApplicationCacheQuirk = false;
```

Comment:
AFAICT we initialize this to false in `QuirksData` and don't modify it anywhere else so is this redundant?